pkgs/by-name/u-/u-config-unwrapped/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  noLibc ? stdenv.cc.isGNU && stdenv.hostPlatform.isx86 && stdenv.hostPlatform.isLinux,
+  pkgsBuildBuild,
+  installShellFiles,
+  nix-update-script,
+}:
+
+let
+  inherit (stdenv) hostPlatform;
+  inherit (lib) optionals;
+
+  platform =
+    if noLibc then
+      (
+        with hostPlatform;
+        if isLinux && isx86_32 then
+          "linux_i686"
+        else if isLinux && isx86_64 then
+          "linux_amd64"
+        else
+          throw "unsupported system ${system} for libc‐free version"
+      )
+    else
+      "generic";
+
+  cflags =
+    [
+      "-DPKG_CONFIG_PREFIX=\"\""
+      "-DPKG_CONFIG_LIBDIR=\"\""
+      "-DPKG_CONFIG_SYSTEM_INCLUDE_PATH=\"\""
+      "-DPKG_CONFIG_SYSTEM_LIBRARY_PATH=\"\""
+    ]
+    ++ optionals noLibc (
+      [
+        "-fno-builtin"
+        "-fno-asynchronous-unwind-tables"
+        "-static"
+        "-nostdlib"
+        "-Wl,--gc-sections"
+      ]
+      ++ optionals (stdenv.cc.isGNU) [
+        # avoid generating calls to memset() and memcpy()
+        "-fno-tree-loop-distribute-patterns"
```

Comment:
Can you explain to me why we need to avoid calling `mem(set|cpy)`? I understand it may cause issues without a libc, but `stdenv.cc.isGNU` assert that we are using glibc, right?

Reply:
The libc‐free versions are not linked to the system libc, but instead perform system calls directly.

They don’t call `memset()` or `memcpy()` themselves, the compiler may however detect memset‐ and memcpy‐like patterns in the code and replace them with calls to those functions.

This is an issue as we are explicitly not linking the system libc with `-nostdlib`.

I have yet to figure out how to avoid that when compiling with Clang / LLVM, but for GCC `-fno-tree-loop-distribute-patterns` should do the trick.